File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> result;
        unordered_map<int,int> mp;
        sort(nums.begin(),nums.end());

        for(int i=nums[0];i<=nums[nums.size()-1];i++){
            mp[i]++;
        }
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        for(auto it: mp){
            if(it.second ==1) result.push_back(it.first);
        }
sort(result.begin(),result.end());
        return result;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        int n=nums.length;
        Arrays.sort(nums);
        Map<Integer,Integer> map=new HashMap<>();
        for(int i=nums[0];i<=nums[n-1];i++){
            map.put(i,0);

        }
        for(int j=0;j<n;j++){
            
            // map.put(nums[j],map.get(nums[j]+1));
              map.put(nums[j], 1);
            
        }
        ArrayList<Integer> arr =new ArrayList<>();
        for(int i=nums[0];i<=nums[n-1];i++){
             if(map.get(i) == 0){
                arr.add(i);
            }
        }

return arr;

    }
}